src/client/serialize/daos_serialize.c

```diff
@@ -0,0 +1,1064 @@
+		type = DAOS_PROP_CO_ENCRYPT;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_ENCRYPT");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_OWNER") > 0) {
+		type = DAOS_PROP_CO_OWNER;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_str(file_id, entry, "DAOS_PROP_CO_OWNER");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_OWNER_GROUP") > 0) {
+		type = DAOS_PROP_CO_OWNER_GROUP;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_str(file_id, entry,
+				     "DAOS_PROP_CO_OWNER_GROUP");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_DEDUP") > 0) {
+		type = DAOS_PROP_CO_DEDUP;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_DEDUP");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_DEDUP_THRESHOLD") > 0) {
+		type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_DEDUP_THRESHOLD");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_ALLOCED_OID") > 0) {
+		type = DAOS_PROP_CO_ALLOCED_OID;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		rc = deserialize_uint(file_id, &entry->dpe_val,
+				      "DAOS_PROP_CO_ALLOCED_OID");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	if (H5Aexists(file_id, "DAOS_PROP_CO_ACL") > 0) {
+		type = DAOS_PROP_CO_ACL;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		entry = &prop->dpp_entries[prop_num];
+		/* read acl as a list of strings in deserialize, then
+		 * convert back to acl for property entry
+		 */
+		rc = deserialize_acl(file_id, entry, "DAOS_PROP_CO_ACL");
+		if (rc != 0) {
+			D_GOTO(out, rc);
+		}
+		prop_num++;
+	}
+	/* deserialize_label stays false if property doesn't exist above */
+	if (deserialize_label) {
+		type = DAOS_PROP_CO_LABEL;
+		prop->dpp_entries[prop_num].dpe_type = type;
+		prop->dpp_entries[prop_num].dpe_str =
+						strdup(label_entry->dpe_str);
+	}
+	*_prop = prop;
+out:
+	/* close container after checking if label exists in pool */
+	if (close_cont)
+		daos_cont_close(coh, NULL);
+	daos_prop_free(label);
+	return rc;
+}
+
+int
+daos_cont_deserialize_props(daos_handle_t poh, char *filename,
+			    daos_prop_t **props, uint64_t *cont_type)
+{
+	int	rc = 0;
+	hid_t	file_id = 0;
+
+	D_PRINT("Reading metadata file: %s\n", filename);
```

Comment:
I think this public function needs a NULL check on filename before dereferencing